apiserver/facades/client/client/filtering.go

```diff
@@ -191,16 +197,26 @@ func unitMatchPort(u *state.Unit, patterns []string) (bool, bool, error) {
 	return matchPortRanges(patterns, portRanges...)
 }
 
-func buildServiceMatcherShims(s *state.Application, patterns ...string) (shims []closurePredicate, _ error) {
+func buildApplicationNameMatcherShims(appName string, patterns ...string) (shims []closurePredicate, _ error) {
 	// Match on name.
 	shims = append(shims, func() (bool, bool, error) {
 		for _, p := range patterns {
-			if strings.ToLower(s.Name()) == strings.ToLower(p) {
+			if strings.ToLower(appName) == strings.ToLower(p) {
 				return true, true, nil
 			}
 		}
-		return false, false, nil
+		return false, true, nil
 	})
+	return shims, nil
+}
+
+func buildServiceMatcherShims(s *state.Application, patterns ...string) (shims []closurePredicate, _ error) {
```

Comment:
Can we change this to buildApplicationMatcherShims? (since we don't have services to match anymore)
and then we should justify why we have ApplicationMatcher and ApplicationNameMatcher.

Reply:
Renamed. and added comments, although code is self-explanatory, that one matcher deals with only matches on the application name, the other looks more closely at the application and its units to find the match. For example, we look at whether the application is exposed...

Reply:
I've kept the renaming of the method as well as its comments, although new, cleaner approach made the need for buildApplicationNameMatcherShims obsolete.